newrelic/resource_newrelic_one_dashboard.go

```diff
@@ -473,6 +480,34 @@ func dashboardWidgetInitialSortingSchemaElem() *schema.Resource {
 	}
 }
 
+func dashboardWidgetDataFormatSchemaElem() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The column name to be sorted",
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Defines the type of the mentioned column",
+				ValidateFunc: validation.StringInSlice([]string{"decimal", "date", "duration", "recent-relative", "custom", "humanized"}, true),
```

Comment:
Does the API return any validation errors? If so, we probably should let the API handle the validation and the provider should surface the API's error message (if it's a good, clear error message).

Reply:
@sanderblue the API does not return a validation error (on not using a valid type), since all of this goes into the `rawConfiguration` attribute of the mutation to create dashboards, which is all a JSON string that ends up not throwing any validation errors.

However, @vinay-newrelic let's not have this set of strings in the validate function, as this would invalidate any requests going out with types other than the ones listed here, if the API were to include more types in future. We anyway are validating the right combinations of `type` with attributes it's supposed to go with such as `format`, `precision`, etc; so we should be okay to get rid of this `ValidateFunc`

Reply:
we can instead just state in the docs that these are the valid values which go with `type`, and customers would be free to use any other type, if they figure out from the API (in the future) that other types are also being taken

Reply:
@sanderblue as @pranav-new-relic mentioned there are no validaton errors from API side and inorder to avoid any disturbances in future from terraform i have removed this hard check for now.

Reply:
Hmm, well since the API doesn't return validation errors, maybe we should keep that validation in to prevent customers from getting a non-helpful error message from the API. 

Generally speaking, we want the API to handle validation, but if an API doesn't provide any clear messaging for validation errors, I think that's when it's okay to have validation checks in our Terraform resource code. 

 This is also a trickier scenario though since it involves the `rawConfiguration` attribute.